CorsixTH/Src/th_gfx_sdl.cpp

```diff
@@ -1044,10 +1057,34 @@ void sprite_sheet::draw_sprite(render_target* pCanvas, size_t iSprite, int iX,
     }
   }
 
+  if ((overlayFlags & thaof_glowing) != 0) {
+    // Let's convert SDL Ticks to Game Ticks
+    uint32_t tick = SDL_GetTicks() / TICKS_MS_TO_USERTICKS_CONVERSION;
+
+    // We want this to vary between 155 -> 205 -> 255.
+    // We're using increments of 10 degrees within the Sin function and
+    // converting them to rad
+    double currentVariation = sin(tick * 10 * M_PI / 180) * 50;
+
+    // Vary from 40 to 80, so we set 60 + Variation
+    err = SDL_SetTextureColorMod(pTexture, 0, 205 + currentVariation, 0);
+    if (err < 0) {
+      throw std::runtime_error(SDL_GetError());
+    }
+  }
+
   SDL_Rect rcSrc = {0, 0, sprite.width, sprite.height};
   SDL_Rect rcDest = {iX, iY, sprite.width, sprite.height};
 
   pCanvas->draw(pTexture, &rcSrc, &rcDest, iFlags);
+
+  if ((overlayFlags & thaof_glowing) != 0) {
```

Comment:
Can this be an else of the above, or is it important for some reason that this is after the draw and the other condition is before the draw?

Reply:
I don't think so, since it's two different things for what we're branching the code.

Ideally we would have some sort of Lifecycle to Draw the Animations and just add Hooks like `onPreRender`

Reply:
@TheCycoONE  An else? condition is the same as at 1060